WalletWasabi/Wallets/Wallet.cs

```diff
@@ -257,6 +262,33 @@ public override async Task StartAsync(CancellationToken cancel)
 		}
 	}
 
+	private async Task PerformFinalSynchronizationAsync(CancellationToken cancel)
+	{
+		while (!cancel.IsCancellationRequested)
+		{
+			try
+			{
+				await PerformWalletSynchronizationAsync(SyncType.NonTurbo, cancel).ConfigureAwait(false);
+				if (LastProcessedFilter is { } lastProcessedFilter)
+				{
+					SetFinalBestHeight(new Height(lastProcessedFilter.Header.Height));
+				}
+
+				break;
+			}
+			catch (InvalidOperationException)
```

Comment:
Shouldn't we catch all exception, too? Otherwise `UnObservedException` can happen.